tools/config/__init__.py

```diff
@@ -610,24 +610,75 @@ def _get_mem_specs(self, memories, cmsis_part, exception_text):
         raise ConfigException(exception_text)
 
-    @property
-    def rom(self):
-        """Get rom information as a pair of start_addr, size"""
+    def get_all_rom_regions(self, all_regions=False):
+        """Get information of all available rom regions in the form of dictionary 
+        {Region: [start_addr, size]}"""
         # Override rom_start/rom_size
         #
         # This is usually done for a target which:
         # 1. Doesn't support CMSIS pack, or
         # 2. Supports TrustZone and user needs to change its flash partition
+        rom_regions = {}
+        # Counter to keep track of ROM memories supported by target
+        rom_active_memories = 0
+        rom_memories = ['IROM1', 'PROGRAM_FLASH', 'IROM2']
         cmsis_part = self._get_cmsis_part()
-        rom_start, rom_size = self._get_mem_specs(
-            ["IROM1", "PROGRAM_FLASH"],
-            cmsis_part,
-            "Not enough information in CMSIS packs to build a bootloader "
-            "project"
-        )
-        rom_start = int(getattr(self.target, "mbed_rom_start", False) or rom_start, 0)
-        rom_size = int(getattr(self.target, "mbed_rom_size", False) or rom_size, 0)
-        return (rom_start, rom_size)
+        for memory in rom_memories:
+            try:
+                rom_start, rom_size = self._get_mem_specs(
+                    memory.split(),
+                    cmsis_part,
+                    "Not enough information in CMSIS packs to build a bootloader "
+                    "project"
+                )
+                rom_start = int(rom_start, 0)
+                rom_size = int(rom_size, 0)
+                if not all_regions:
+                    return rom_start, rom_size
+                if memory=='IROM1' or memory=='PROGRAM_FLASH':
+                    memory = 'ROM'
+                else:
+                    rom_active_memories += 1
+                    memory = 'ROM' + str(rom_active_memories)
+                rom_regions[memory] = [rom_start, rom_size]
+            except ConfigException:
+                continue         
+        return rom_regions
+
+    def get_all_ram_regions(self):
+        """Get information of all available ram regions in the form of dictionary 
+        {Region: [start_addr, size]}"""
+        # Override rom_start/rom_size
+        #
+        # This is usually done for a target which:
+        # 1. Doesn't support CMSIS pack, or
+        # 2. Supports TrustZone and user needs to change its flash partition
+        ram_regions = {}
+        # Counter to keep track of RAM memories supported by target
+        ram_active_memories = 0
+        ram_memories = ['IRAM1', 'IRAM2', 'IRAM3', 'IRAM4', 'SRAM_OC', \
+                        'SRAM_ITC', 'SRAM_DTC', 'SRAM_UPPER', 'SRAM_LOWER']
+        cmsis_part = self._get_cmsis_part()
+        for memory in ram_memories:
+            try:
+                ram_start, ram_size = self._get_mem_specs(
+                    memory.split(),
```

Comment:
Same question about `.split()` here. If it's just to turn a string into an array I'd recommend doing that explicitly:

```suggestion
                    [memory],
```